kratos/processes/vtk_output_process.cpp

```diff
@@ -0,0 +1,734 @@
+
+    for (ModelPart::ElementIterator elem_i = model_part.ElementsBegin(); elem_i != model_part.ElementsEnd(); ++elem_i)
+    {
+        vtk_cell_list_size++;
+        vtk_cell_list_size += elem_i->GetGeometry().size();
+    }
+
+    for (ModelPart::ConditionIterator condition_i = model_part.ConditionsBegin(); condition_i != model_part.ConditionsEnd(); ++condition_i)
+    {
+        vtk_cell_list_size++;
+        vtk_cell_list_size += condition_i->GetGeometry().size();
+    }
+
+    return vtk_cell_list_size;
+}
+
+void VtkOutputProcess::Initialize(ModelPart &model_part)
+{
+    CreateMapFromKratosIdToVTKId(model_part);
+    mVtkCellListSize = DetermineVtkCellListSize(model_part);
+}
+
+void VtkOutputProcess::WriteHeader(ModelPart &model_part)
+{
+    std::string outputFileName = GetOutputFileName(model_part);
+    std::ofstream outputFile;
+
+    outputFile.open(outputFileName, std::ios::out | std::ios::binary | std::ios::trunc);
+    outputFile << "# vtk DataFile Version 4.0"
+               << "\n";
+    outputFile << "vtk output"
+               << "\n";
+    outputFile << "ASCII"
+               << "\n";
+    outputFile << "DATASET UNSTRUCTURED_GRID"
+               << "\n";
+    outputFile.close();
+}
+
+void VtkOutputProcess::WriteMesh(ModelPart &model_part)
+{
+    WriteNodes(model_part);
+    WriteConditionsAndElements(model_part);
+    WriteConditionAndElementTypes(model_part);
+}
+
+void VtkOutputProcess::WriteNodes(ModelPart &model_part)
+{
+    std::string outputFileName = GetOutputFileName(model_part);
+    std::ofstream outputFile;
+    outputFile.open(outputFileName, std::ios::out | std::ios::app | std::ios::binary);
+    outputFile << std::scientific;
+    outputFile << std::setprecision(mDefaultPrecision);
+
+    // write nodes header
+    outputFile << "POINTS " << model_part.NumberOfNodes() << " float"
+               << "\n";
+
+    // write nodes
+    for (ModelPart::NodeIterator node_i = model_part.NodesBegin(); node_i != model_part.NodesEnd(); ++node_i)
+    {
+        double x_coordinate = node_i->X();
+        double y_coordinate = node_i->Y();
+        double z_coordinate = node_i->Z();
+        outputFile << " " << x_coordinate;
+        outputFile << " " << y_coordinate;
+        outputFile << " " << z_coordinate << "\n";
+    }
+
+    outputFile.close();
+}
+
+void VtkOutputProcess::WriteConditionsAndElements(ModelPart &model_part)
+{
+    std::string outputFileName = GetOutputFileName(model_part);
+    std::ofstream outputFile;
+    outputFile.open(outputFileName, std::ios::out | std::ios::app | std::ios::binary);
+
+    // write cells header
+    outputFile << "CELLS " << model_part.NumberOfConditions() + model_part.NumberOfElements() << " " << mVtkCellListSize << "\n";
+
+    // write elements
+    for (ModelPart::ElementIterator elem_i = model_part.ElementsBegin(); elem_i != model_part.ElementsEnd(); ++elem_i)
+    {
+        ModelPart::ConditionType::GeometryType &elem_geometry = elem_i->GetGeometry();
```

Comment:
```suggestion
        auto& elem_geometry = elem_i->GetGeometry();
```

To avoid large lines